### PR TITLE
Add ManagedApp to ZIO

### LINKED ISSUE #656: Add ManagedApp to ZIO
Like `scalaz.zio.App`, but `main` should return `Managed[Environment, Nothing, Int]`.

### PR DESCRIPTION
**Pull request by bhop**
fixes #656

### CODE REVIEW

File: core/jvm/src/main/scala/scalaz/zio/ManagedApp.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalaz.zio
+
+trait ManagedApp extends DefaultRuntime {
```
Comment (iravid) [inline core/jvm/src/main/scala/scalaz/zio/ManagedApp.scala]: Thank you for working on this! Would it be possible to reuse the existing App so we won’t have to duplicate the implementation?

Comment (bhop) [reply core/jvm/src/main/scala/scalaz/zio/ManagedApp.scala]: Yep, make sense. Thank you.

Comment (iravid) [reply core/jvm/src/main/scala/scalaz/zio/ManagedApp.scala]: Yeah you don't have to override it. You can keep an instance of `App` as a member in this trait, and delegate to its main method from our main method.

Comment (bhop) [reply core/jvm/src/main/scala/scalaz/zio/ManagedApp.scala]: Yeah, you are right. Thanks!

**Pull request merged**